Fix `qml.prod` when qfunc contains a single operator (#5170)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Bug fixes 🐛</h3>

> * Fixes a queuing bug when taking the `prod` of a qfunc that queues a single operator.

Author: ```suggestion
* Fixes a queuing bug when taking the `prod` of a qfunc that queues a single operator.
  [(#5170)](https://github.com/PennyLaneAI/pennylane/pull/5170)
```

<h3>Bug fixes 🐛</h3>

> * Fixes a queuing bug when taking the `prod` of a qfunc that queues a single operator.

Reviewer: ```suggestion
* Fixes a queuing bug when using `qml.prod` with a qfunc that queues a single operator.
```
